tools/rostopic/src/rostopic/__init__.py

```diff
@@ -173,40 +176,95 @@ def print_hz(self):
+
+def _rostopic_hz(topics, window_size=-1, filter_expr=None):
     """
     Periodically print the publishing rate of a topic to console until
     shutdown
-    :param topic: topic name, ``str``
+    :param topics: topic names, ``list`` of ``str``
     :param window_size: number of messages to average over, -1 for infinite, ``int``
     :param filter_expr: Python filter expression that is called with m, the message instance
     """
-    msg_class, real_topic, _ = get_topic_class(topic, blocking=True) #pause hz until topic is published
     if rospy.is_shutdown():
         return
     rospy.init_node(NAME, anonymous=True)
-    rt = ROSTopicHz(window_size, filter_expr=filter_expr)
-    # we use a large buffer size as we don't know what sort of messages we're dealing with.
-    # may parameterize this in the future
-    if filter_expr is not None:
-        # have to subscribe with topic_type
-        sub = rospy.Subscriber(real_topic, msg_class, rt.callback_hz)
-    else:
-        sub = rospy.Subscriber(real_topic, rospy.AnyMsg, rt.callback_hz)        
-    print("subscribed to [%s]"%real_topic)
+    hz_checkers = []
+    for topic in topics:
+        msg_class, real_topic, _ = get_topic_class(topic, blocking=True) # pause hz until topic is published
+        rt = ROSTopicHz(window_size, filter_expr=filter_expr)
+        # we use a large buffer size as we don't know what sort of messages we're dealing with.
+        # may parameterize this in the future
+        if filter_expr is not None:
+            # have to subscribe with topic_type
+            rospy.Subscriber(real_topic, msg_class, rt.callback_hz)
+        else:
+            rospy.Subscriber(real_topic, rospy.AnyMsg, rt.callback_hz)
+        hz_checkers.append(rt)
+        print("subscribed to [%s]" % real_topic)
 
     if rospy.get_param('use_sim_time', False):
         print("WARNING: may be using simulated time",file=sys.stderr)
 
     while not rospy.is_shutdown():
         _sleep(1.0)
-        rt.print_hz()
+        if len(topics) == 1:
+            hz_checkers[0].print_hz()
+            continue
+        # monitoring multiple topics' hz
+        header = ['topic', 'rate', 'min_delta', 'max_delta', 'std_dev', 'window']
+        max_col_width = [len(h) for h in header]
+        stats = []
+        if not any(rt.times for rt in hz_checkers):
+            continue  # wait for initial message
+        for topic, rt in zip(topics, hz_checkers):
+            hz_stat = rt.get_hz()
+            if hz_stat is None:
+                continue
+            rate, min_delta, max_delta, std_dev, window = hz_stat
+            stat = [topic, '{:.4}'.format(rate), '{:.4}'.format(min_delta),
+                    '{:.4}'.format(max_delta), '{:.4}'.format(std_dev),
+                    str(window)]
+            stats.append(stat)
+            for i, s in enumerate(stat):
+                col_width = len(s)
+                if col_width > max_col_width[i]:
+                    max_col_width[i] = col_width
```

Comment:
These computation are all overly complicated. Since you have to manually iterate over all cells in a single column to determine the maximum width (including the header) I think it would make much more sense to store all cells of a column in a list (rather then the cells of a row). That way the majority of this code becomes obsolete.